turbinia/turbiniactl.py

```diff
@@ -854,40 +856,28 @@ def main():
     request = TurbiniaRequest(
         request_id=request_id, requester=getpass.getuser())
     request.evidence.append(evidence_)
-    if filter_patterns:
-      request.recipe['filter_patterns'] = filter_patterns
-    if args.jobs_denylist:
-      request.recipe['jobs_denylist'] = args.jobs_denylist
-    if args.jobs_allowlist:
-      request.recipe['jobs_allowlist'] = args.jobs_allowlist
-    if yara_rules:
-      request.recipe['yara_rules'] = yara_rules
-    if args.debug_tasks:
-      request.recipe['debug_tasks'] = args.debug_tasks
-    if args.recipe_config:
-      for pair in args.recipe_config:
-        try:
-          key, value = pair.split('=')
-        except ValueError as exception:
-          log.error(
-              'Could not parse key=value pair [{0:s}] from recipe config '
-              '{1!s}: {2!s}'.format(pair, args.recipe_config, exception))
-          sys.exit(1)
-        request.recipe[key] = value
-    if args.decryption_keys:
-      for credential in args.decryption_keys:
-        try:
-          credential_type, credential_data = credential.split('=')
-        except ValueError as exception:
-          log.error(
-              'Could not parse credential [{0:s}] from decryption keys '
-              '{1!s}: {2!s}'.format(
-                  credential, args.decryption_keys, exception))
-          sys.exit(1)
-        evidence_.credentials.append({
-            'credential_type': credential_type,
-            'credential_data': credential_data
-        })
+
+    if args.recipe:
+      if (args.jobs_denylist or args.jobs_allowlist or
+          args.filter_patterns_file or args.yara_rules_file):
+        raise TurbiniaException(
+            'Specifying a recipe is incompatible with defining'
+            ' jobs allow/deny lists, yara rules or a patterns file separately.')
+        sys.exit(1)
+      recipe_file = os.path.join(config.RECIPE_FILE_DIR, args.recipe)
+      recipe_dict = recipe_helpers.load_recipe_from_file(recipe_file)
+      if not recipe_dict:
+        sys.exit(1)
+    else:
+      recipe_dict = copy.deepcopy(recipe_helpers.DEFAULT_RECIPE)
+
+      recipe_dict['globals']['jobs_denylist'] = args.jobs_denylist
+      recipe_dict['globals']['jobs_allowlist'] = args.jobs_denylist
+      recipe_dict['globals']['yara_rules_file'] = args.jobs_denylist
+      recipe_dict['globals']['filter_patterns_file'] = args.jobs_denylist
+
+    request.recipe['task_recipes'] = recipe_dict
```

Comment:
Can we flatten this a bit and get rid of the outer dict that will only have the one 'task_recipes' key (ie. just set `request.recipe = recipe_dict`?  That will also allow us to simplify/remove some of the other methods in the task_manager, etc.   I know this is allowing for the potential of job_recipes in the future, but I think we should find a way to continue to keep that in a flatter dictionary similar to how it's defined in the YAML.  We may end up still expanding things later, but that's ok.

Reply:
this looks doable. should we move the "requester" to the globals recipe? Can't see anywhere else where the request recipe is edited.

Reply:
Done tentatively in this manner.